### ISSUE TITLE
E: no-group-tag check is never executed

### ISSUE THREAD

**Issue by thisisshub** [rpm-software-management/rpmlint#611]
## Source of the Issue

https://github.com/rpm-software-management/rpmlint/blob/60c98ff0f92e79ddcc224088f7a0c4dc36416c55/rpmlint/checks/TagsCheck.py#L479-L492

In line 491 above:
```python
if not group:
```

The `group` variable originates from line 93 of TagsCheck,py `group = pkg[rpm.RPMTAG_GROUP]` which basically returns the value of the `Group:` tag in a specfile and does not check for the `Group:` tag itself.

## Experimentation

Creating a test.spec with a random `Group:` tag value then making a binary of it and finally testing it with `rpmlint` 

<details>
<summary>test.spec</summary>
<br>

```specfile
Name:           test
Version:        0
Release:        0
Summary:        non-standard-group warning
License:        Apache License
Group:          some/group/value

%description

%prep

%build

%install

%files

%changelog
```
</details>

<details>
<summary>TagsCheck.py</summary>
<br>

```python
# Check if a package does not have a group tag
print("=======")
print(group)
print("=======")
if not group:
    self.output.add_info('E', pkg, 'no-group-tag')
```

</details>

Upon running

```bash
$ rpmbuild -bb test.spec
$ rpmlint /home/shub/rpmbuild/RPMS/x86_64/test-0-0.x86_64.rpm
```

**Expected Result**

```bash
===========
Group:
===========
============================ rpmlint session starts ============================
rpmlint: 2.0.0
configuration:
    /home/shub/gsoc/rpmlint/rpmlint/configdefaults.toml
```

**Actual Result**

```bash
===========
some/group/value
===========
============================ rpmlint session starts ============================
rpmlint: 2.0.0
configuration:
    /home/shub/gsoc/rpmlint/rpmlint/configdefaults.toml
```

**Comment by thisisshub** [comment]
Since the `if not group:` tag checks or looks for the `Group:` tag's value and not for the tag itself the key note problem becomes that the `if not group:` condition is never executed perhaps.

OTOH if we do remove the `Group:` tag we are finally return with the following result

<details>
<summary>test.spec without Group tag</summary>
<br>

```specfile
Name:           test
Version:        0
Release:        0
Summary:        non-standard-group warning
License:        Apache License

%description

%prep

%build

%install

%files

%changelog
```

</details>

**Output**
```bash
$ rpmlint /home/shub/rpmbuild/RPMS/x86_64/test-0-0.x86_64.rpm

===========
Unspecified
===========
============================ rpmlint session starts ============================
rpmlint: 2.0.0
configuration:
    /home/shub/gsoc/rpmlint/rpmlint/configdefaults.toml
```

**Comment by marxin** [comment]
Confirmed, I've just created a pull request.
The corresponding `rpm` code live here:

https://github.com/rpm-software-management/rpm/blob/d8fbddfa5051bdc1c71e16cb11f14d9fdc7f5c5e/build/parsePreamble.c#L1206-L1212
